drivers/hmis/app/models/hmis/hud/enrollment.rb

```diff
@@ -336,6 +336,58 @@ def data_collection_features
     end.compact
   end
 
+  def occurrence_point_forms
+    # All instances for Occurrence Point forms, including inactive/retired, so we can check for legacy data.
+    # Draft forms should never have legacy data, but let's include them just in case.
+    base_scope = Hmis::Form::Instance.with_role(:OCCURRENCE_POINT)
+
+    # All possible form identifiers used for Occurrence Point collection
+    occurrence_point_identifiers = base_scope.pluck(:definition_identifier).uniq
+
+    # Get the latest version of each form definition so we can check for legacy data
+    definitions = Hmis::Form::Definition.where(identifier: occurrence_point_identifiers).latest_versions
+
+    definitions.map do |definition|
+      # Choose the most specific instance for each definition identifier
+      instance_scope = base_scope.where(definition_identifier: definition.identifier).order(updated_at: :desc)
+      best_instance = instance_scope.active.published.detect_best_instance_for_enrollment(enrollment: self)
```

Comment:
I'm not following why we need `published` which does another join back to the definition table- can you explain that one? (My suggestion above excludes it, but maybe there is something I'm missing)